clang/lib/CodeGen/HLSLBufferLayoutBuilder.h

```diff
@@ -25,24 +24,34 @@ class CodeGenModule;
 class HLSLBufferLayoutBuilder {
 private:
   CodeGenModule &CGM;
-  llvm::StringRef LayoutTypeName;
 
 public:
-  HLSLBufferLayoutBuilder(CodeGenModule &CGM, llvm::StringRef LayoutTypeName)
-      : CGM(CGM), LayoutTypeName(LayoutTypeName) {}
-
-  // Returns LLVM target extension type with the name LayoutTypeName
-  // for given structure type and layout data. The first number in
-  // the Layout is the size followed by offsets for each struct element.
-  llvm::TargetExtType *createLayoutType(const RecordType *StructType,
-                                        const CGHLSLOffsetInfo &OffsetInfo);
-
-private:
-  bool layoutField(const clang::FieldDecl *FD, unsigned &EndOffset,
-                   unsigned &FieldOffset, llvm::Type *&FieldType,
-                   uint32_t Packoffset);
-  bool layoutField(const clang::FieldDecl *FD, unsigned &EndOffset,
-                   unsigned &FieldOffset, llvm::Type *&FieldType);
+  HLSLBufferLayoutBuilder(CodeGenModule &CGM) : CGM(CGM) {}
+
+  /// Lays out a struct type following HLSL buffer rules and considering
+  /// PackOffsets, if provided. Previously created layout structs are cached by
+  /// CGHLSLRuntime.
+  ///
+  /// The function iterates over all fields of the record type (including base
+  /// classes) and calls layoutField to converts each field to its corresponding
+  /// LLVM type and to calculate its HLSL constant buffer layout. Any embedded
+  /// structs (or arrays of structs) are converted to layout types as well.
+  ///
+  /// When PackOffsets are specified the elements will be placed based on the
+  /// user-specified offsets. Not all elements must have a
+  /// packoffset/register(c#) annotation though. For those that don't, the
+  /// PackOffsets array will contain -1 value instead. These elements must be
+  /// placed at the end of the layout after all of the elements with specific
+  /// offset.
```

Comment:
This comment needs to be updated. It talks about PackOffsets, but the code uses OffsetInfo. And `layoutField` function does not exist.